app/jsx/components/DrawerComp.jsx

```diff
@@ -40,24 +40,38 @@ class SortableNavList extends React.Component {
   }
 
   generateData(navItems) {
-    if (!navItems)
-      return undefined
-    return navItems.map(nav => {
-      let data = {
-        id: nav.id,
-        type: nav.type,
-        slug: nav.slug,
-        title: <Link to={`/uc/${this.props.unit}${(nav.type == "home") ? "" : `/nav/${nav.id}`}`}>{nav.name}</Link>,
-        subtitle: <i>{nav.hidden && 'hidden '}{nav.type}</i>
-      }
-      if (nav.type == "folder") {
-        data.children = this.generateData(nav.sub_nav)
-        data.expanded = true
-      }
-      else
-        data.noChildren = true
-      return data
-    })
+    let generate = (navItems) => {
+      if (!navItems)
+        return undefined
+      return navItems.map(nav => {
+        let data = {
+          id: nav.id,
+          type: nav.type,
+          slug: nav.slug,
+          title: <Link to={`/uc/${this.props.unit}${(nav.type.includes("fixed")) ? "" : `/nav/${nav.id}`}`}>{nav.name}</Link>,
+          subtitle: <i>{nav.hidden && 'hidden '}{nav.type.replace(/_/g," ")}</i>
+        }
+        if (nav.type.includes("folder") && (nav.id >= 0 || nav.id == -9999)) {
```

Comment:
This could use a comment. I think I know what the >= 0 || == -9999 is about (avoiding the issues sub-nav) but I had to really think about it.